controllers/topology/desired_state.go

```diff
@@ -170,6 +182,106 @@ func computeCluster(current *clusterTopologyState, infrastructureCluster, contro
 	return cluster
 }
 
+// computeMachineDeployment computes the desired state for a MachineDeploymentTopology.
+// The generated machineDeployment object is calculated using the values from the machineDeploymentTopology and
+// the machineDeployment class.
+func computeMachineDeployment(class *clusterTopologyClass, current *clusterTopologyState, topology clusterv1.MachineDeploymentTopology) (machineDeploymentTopologyState, error) {
+	desiredMachineDeployment := machineDeploymentTopologyState{}
+
+	className := topology.Class
+	machineDeploymentClass, err := getMachineDeploymentClass(class, className)
+	if err != nil {
+		return desiredMachineDeployment, err
+	}
+
+	machineDeploymentTopologyClass, ok := class.machineDeployments[className]
+	if !ok {
+		return desiredMachineDeployment, errors.Errorf("class name %s not found in clusterClass %s", className, class.clusterClass.Name)
+	}
```

Comment:
This is a little bit confusing because we are going to the class two times to take informations about the same thing, and also it forces us to introduce `machineDeploymentTopologyClass` which is a confusing middle ground between `machineDeploymentClass` and (machineDeployment)`topology`.

What if we add metadata to machineDeploymentTopologyClass, so we can get everything in a single place?

```
type machineDeploymentTopologyClass struct {
        metadata ObjectMeta `json:"metadata,omitempty"`
	bootstrapTemplate             *unstructured.Unstructured
	infrastructureMachineTemplate *unstructured.Unstructured
}
```
(this requires a small change to getClass)